src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -2960,7 +2960,7 @@ internal Conversion ClassifyConversionForCast(int position, ExpressionSyntax exp
         /// <param name="declarationSyntax">The syntax node that declares a member.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>The symbol that was declared.</returns>
-        public abstract ISymbol GetDeclaredSymbol(LocalFunctionStatementSyntax declarationSyntax, CancellationToken cancellationToken = default(CancellationToken));
+        public abstract IMethodSymbol GetDeclaredSymbol(LocalFunctionStatementSyntax declarationSyntax, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
Huh. I was going to suggest that we should update the public API for this as well, but we apparently don't have a `GetDeclaredSymbol` extension for local functions. That seems like a bug in itself, as it means that https://github.com/dotnet/roslyn-analyzers/pull/6779 is going to warn when calling `GetDeclaredSymbol` on a local function syntax node.

Reply:
And now I see that you're driving this because of a request for just such an API in https://github.com/dotnet/roslyn/issues/71028 😆.

Reply:
yup :D